elf/elf.cc

```diff
@@ -103,8 +103,14 @@ elf::elf(const std::shared_ptr<loader> &l)
         // Load sections
         const void *sec_data = l->load(m->hdr.shoff,
                                        m->hdr.shentsize * m->hdr.shnum);
+        if (NULL == sec_data) {
```

Comment:
loader.load is not allowed to return NULL (it's supposed to throw an exception if there's a problem) and the mmap loader implementation doesn't return NULL, so I'm afraid I don't understand what this change is accomplishing.